phpBB/phpbb/db/extractor/base_extractor.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\db\extractor;
+
+abstract class base_extractor implements extractor_interface
+{
+	/**
+	 * @var    string    phpBB root path
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var    \phpbb\request\request_interface
+	 */
+	protected $request;
+
+	/**
+	 * @var    \phpbb\db\driver\driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var    bool
+	 */
+	protected $download;
+
+	/**
+	 * @var    bool
+	 */
+	protected $store;
+
+	/**
+	 * @var    int
+	 */
+	protected $time;
+
+	/**
+	 * @var    string
+	 */
+	protected $format;
+
+	/**
+	 * @var    resource
+	 */
+	protected $fp;
+
+	/**
+	 * @var string
+	 */
+	protected $write;
+
+	/**
+	 * @var string
+	 */
+	protected $close;
+
+	/**
+	 * @var bool
+	 */
+	protected $run_comp;
+
+	/**
+	 * Constructor
+	 *
+	 * @param string $phpbb_root_path
+	 * @param \phpbb\request\request_interface $request
+	 * @param \phpbb\db\driver\driver_interface $db
+	 */
+	public function __construct($phpbb_root_path, \phpbb\request\request_interface $request, \phpbb\db\driver\driver_interface $db)
+	{
+		$this->phpbb_root_path	= $phpbb_root_path;
+		$this->request			= $request;
+		$this->db				= $db;
+		$this->fp				= null;
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function init_extractor($format, $filename, $time, $download = false, $store = false)
+	{
+		$this->download			= $download;
+		$this->store			= $store;
+		$this->time				= $time;
+		$this->format			= $format;
+
+		switch ($format)
+		{
+			case 'text':
+				$ext = '.sql';
+				$open = 'fopen';
+				$this->write = 'fwrite';
+				$this->close = 'fclose';
+				$mimetype = 'text/x-sql';
+				break;
```

Comment:
break; should have the same indentation as case. See: http://area51.phpbb.com/docs/31x/coding-guidelines.html#codelayout "Switch statements"